docs/design/replacement_buckets.md

```diff
@@ -0,0 +1,135 @@
+# Replacement Buckets For Automatic Replacements
+
+## Metadata
+
+* Authors: @johscheuer
+* Created: 2022-10-03
+* Updated: 2022-10-03
+
+## Background
+
+The operator supports to replace process groups with different conditions like `MissingProcess`.
+This allows the operator to bring the cluster into a better state if some process groups are failing.
+A human operator can currently limit the number of automatic replacements, by default the operator will only replace one process group at a time.
+The drawback of this approach is that all replacements will be accounted in the same way.
+That brings the risk that some replacements are blocked for a longer time by a single (or small) number of replacements.
+Replacements for stateless or log process groups are typically fast, but a single replacement of a storage process group could block those replacements for a long time.
+
+## General Design Goals
+
+Allow a human operator to define replacement buckets which allow to set limits on a per process class base.
```

Comment:
does operator have a safeguard to prevent human error (e.g., set the limits too high?)